packages/synthetic-chain/src/cli/doctor.ts

```diff
@@ -0,0 +1,68 @@
+import fs from 'node:fs';
+import path from 'node:path';
+import { ProposalInfo } from './proposals.ts';
+import assert from 'node:assert';
+import { execSync } from 'node:child_process';
+
+const fixupProposal = (proposal: ProposalInfo) => {
+  const proposalPath = path.join(
+    'proposals',
+    `${proposal.proposalIdentifier}:${proposal.proposalName}`,
+  );
+  console.log('checking proposal', proposalPath, '...');
+  const packageJson = JSON.parse(
+    fs.readFileSync(path.join(proposalPath, 'package.json'), 'utf-8'),
+  );
+  if (packageJson.dependencies) {
+    assert(
+      packageJson.packageManager,
+      'missing packageManager in package.json',
+    );
+    if (packageJson.packageManager.includes('yarn')) {
+      console.log('found "yarn" packageManager, processing...');
+      // ensure it has its own yarn.lock
+      fs.writeFileSync(
+        path.join(proposalPath, 'yarn.lock'),
+        JSON.stringify(packageJson, null, 2),
+      );
```

Comment:
write the `package.json` contents to `yarn.lock`? that doesn't seem right. Am I reading this right?

Reply:
you are, it's a bug. I thought I changed this to write ''.